src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -678,8 +686,24 @@ class DottyBackendInterface()(implicit ctx: Context) extends BackendInterface{
 
     // members
     def primaryConstructor: Symbol = toDenot(sym).primaryConstructor
-    def nestedClasses: List[Symbol] = memberClasses //exitingPhase(currentRun.lambdaliftPhase)(sym.memberClasses)
-    def memberClasses: List[Symbol] = toDenot(sym).info.memberClasses.map(_.symbol).toList
+
+    /** For currently compiled classes: All locally defined classes including local classes.
```

Comment:
Yes, "local classes" is Java lingo.

On Fri, May 8, 2015 at 4:43 PM, Dmitry Petrashko <notifications@github.com>
wrote:

> In src/dotty/tools/backend/jvm/DottyBackendInterface.scala
> <https://github.com/lampepfl/dotty/pull/543#discussion_r29944973>:
>
> > @@ -678,8 +686,24 @@ class DottyBackendInterface()(implicit ctx: Context) extends BackendInterface{
> >
> >      // members
> >      def primaryConstructor: Symbol = toDenot(sym).primaryConstructor
> > -    def nestedClasses: List[Symbol] = memberClasses //exitingPhase(currentRun.lambdaliftPhase)(sym.memberClasses)
> > -    def memberClasses: List[Symbol] = toDenot(sym).info.memberClasses.map(_.symbol).toList
> > +
> > +    /** For currently compiled classes: All locally defined classes including local classes.
>
> My guess is:
>
> class A{
>   class B
>   def foo = {
>     class C
>   }
> }
>
> Class B is locally defined, class C is local.
> Though the naming here is indeed confusing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lampepfl/dotty/pull/543/files#r29944973>.
>



-- 
Martin Odersky
EPFL